ql/src/test/org/apache/hadoop/hive/ql/exec/PartitionUtil.java

```diff
@@ -82,4 +88,26 @@ public static Set<CheckResult.PartitionResult> createPartsNotInMs(int numOfParts
         }
         return partsNotInMs;
     }
+
+
+    public static void addPartitions(IMetaStoreClient db, String dbName, String tableName, String location,
+         HiveConf hiveConf, int numPartitions) throws Exception {
+        List<Partition> partitions = new ArrayList<>();
+        for (int i = 0; i < numPartitions; i++) {
+            partitions.add(buildPartition(dbName, tableName, String.valueOf(i), location + "/city=" + i, hiveConf));
+        }
+        db.add_partitions(partitions, true, true);
+    }
+
+    protected static Partition buildPartition(String dbName, String tableName, String value,
```

Comment:
why it's protected

Reply:
It is as a part of refactoring that i moved it to this class as it was being used in multiple places. If any change is required for this, i can do it as a part of a follow-up JIRA. Let me know if it required.